pkgs/data/misc/iana-etc/default.nix

```diff
@@ -1,17 +1,83 @@
-{stdenv, fetchurl}:
+{ stdenv, fetchurl }:
 
-stdenv.mkDerivation rec {
-  name = "iana-etc-2.30";
+let
 
-  src = fetchurl {
-    url = "http://sethwklein.net/${name}.tar.bz2";
-    sha256 = "03gjlg5zlwsdk6qyw3v85l129rna5bpm4m7pzrp864h0n97qg9mr";
+  iana = {
+    
+    protocolsXML = fetchurl {
+      urls = [
+        "https://www.iana.org/assignments/protocol-numbers/protocol-numbers.xml"
+        "ftp://ftp.iana.org/assignments/protocol-numbers/protocol-numbers.xml"
+      ];
+      sha256 = "f7a7fd69b66b936cdcba4416066cba8d8e88b6261ba43c2c2fab8cc3c6876825";
+    };
+    
+    servicesXML = fetchurl {
+      urls = [
+        "https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xml"
+        "ftp://ftp.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xml"
+      ];
+      sha256 = "328d0d3349646757d78450bc0edee35120522553debc2b5cf537614f33e500fe";
+    };
+    
   };
 
-  preInstall = "installFlags=\"PREFIX=$out\"";
+in
+stdenv.mkDerivation rec {
+
+  name = "iana-etc-${version}";
+  version = "20170307";
+  srcs = [ iana.protocolsXML iana.servicesXML ];
 
+  
+  protocolsGawk = ''
+    BEGIN {
+      print "# See also protocols(5) and IANA official page :"
+      print "# https://www.iana.org/assignments/protocol-numbers \n#"
+      FS="[<>]"
+    }
+    {
+      if (/<updated/) { print "# Last updated: " $3 "\n" ; next }
+      if (/<record/) { v=n=0 }
+      if (/<value/) v=$3
+      if (/<name/ && !($3~/ /)) n=$3
+      if (/<\/record/ && (v || n=="HOPOPT") && n) printf "%-16s %3i %s\n", tolower(n),v,n
+      if (/<people/) exit
+    }
+  '';
+  
+  servicesGawk = ''
+    BEGIN {
+      print "# See also services(5) and IANA official page :"
+      print "# https://www.iana.org/assignments/service-names-port-numbers \n#"
+      FS="[<>]"
+    }
+    {
+      if (/<updated/) { print "# Last updated: " $3 "\n" ; next }
+      if (/<record/) { n=u=p=c=0 }
+      if (/<name/ && !/\(/) n=$3
+      if (/<number/) u=$3
+      if (/<protocol/) p=$3
+      if (/Unassigned/ || /Reserved/ || /historic/) c=1
+      if (/<\/record/ && n && u && p && !c) printf "%-16s %5i/%s\n", n,u,p
+      if (/<people/) exit
+    }
+  '';
+  
+  
+  phases = [ "installPhase" ];
+  buildInputs = [ stdenv ];
```

Comment:
It shouldn't be; anything that's part of the `stdenv` is added to the build environment implicitly. Adding `stdenv` to inputs is like asking to extend the environment with itself, which at least conceptually is the identity operation (not sure if that's the case in practice).